openlibrary/core/search.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import absolute_import
+
+import urllib
+import urllib2
+import web
+import simplejson
+import six
+
+from infogami.utils import delegate
+
+from openlibrary.utils import dateutil
+from openlibrary.core.models import Edition
+from openlibrary.core.helpers import get_coverstore_url
+from openlibrary.core.helpers import bookreader_host
+from openlibrary.core.lending import config_http_request_timeout
+
+
+PRESET_QUERIES = {
+    'preset:thrillers': '(creator:"Clancy, Tom" OR creator:"King, Stephen" OR creator:"Clive Cussler" OR creator:("Cussler, Clive") OR creator:("Dean Koontz") OR creator:("Koontz, Dean") OR creator:("Higgins, Jack")) AND !publisher:"Pleasantville, N.Y. : Reader\'s Digest Association" AND languageSorter:"English"',
+    'preset:children': '(creator:("parish, Peggy") OR creator:("avi") OR title:("goosebumps") OR creator:("Dahl, Roald") OR creator:("ahlberg, allan") OR creator:("Seuss, Dr") OR creator:("Carle, Eric") OR creator:("Pilkey, Dav"))',
+    'preset:comics': '(subject:"comics" OR creator:("Gary Larson") OR creator:("Larson, Gary") OR creator:("Charles M Schulz") OR creator:("Schulz, Charles M") OR creator:("Jim Davis") OR creator:("Davis, Jim") OR creator:("Bill Watterson") OR creator:("Watterson, Bill") OR creator:("Lee, Stan"))',
+    'preset:authorsalliance_mitpress': '(openlibrary_subject:(authorsalliance) OR collection:(mitpress) OR publisher:(MIT Press) OR openlibrary_subject:(mitpress)) AND (!loans__status__status:UNAVAILABLE)'
+}
+
+# Advanced Search internal calls use http for debugging
+BASE_URL = bookreader_host() or 'archive.org'
+SEARCH_URL = 'http://%s/advancedsearch.php' % BASE_URL
+BROWSE_URL = 'https://%s/search.php' % BASE_URL
+
+AVAILABILITY_STATUS = 'loans__status__status'
+RETURN_FIELDS = ['identifier', AVAILABILITY_STATUS,
+                 'openlibrary_edition', 'openlibrary_work']
+MAX_IA_RESULTS = 100
```

Comment:
This seems like a lot, particularly if it influences latency for results.

Reply:
@tfmorris agreed -- maybe we can make this 50. most front-end components (see refactor/everything) actually limit this to 14 in their calls. @cdrini had a similar comment before in #2151

Reply:
switching to 20 to be consistent w/ #2151